### PR TITLE
New Resource: `azurerm_mssql_job_credential`

### PR DESCRIPTION
**Pull request by r0bnet**
Job Credential resource as part of the Elastic Job Agents.

see: GH-11248

### CODE REVIEW

File: azurerm/internal/services/mssql/mssql_job_credential_resource.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+func resourceMsSqlJobCredentialCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MSSQL.JobCredentialsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for Job Credential creation.")
+
+	name := d.Get("name").(string)
+	jobAgentId := d.Get("job_agent_id").(string)
+	jaId, _ := parse.JobAgentID(jobAgentId)
```
Comment (jackofallops) [inline azurerm/internal/services/mssql/mssql_job_credential_resource.go:75]: Can we make use of the generated ID Function here:
```suggestion
	jaId, err := parse.JobAgentID(d.Get("job_agent_id").(string))
	if err != nil {
		return err
	}
	jobCredentialId := parse.NewJobCredentialID(jaId.SubscriptionId, jaId.ResourceGroup, jaId.ServerName, jaId.Name, d.Get("name").(string))

```

File: azurerm/internal/services/mssql/mssql_job_credential_resource.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	username := d.Get("username").(string)
+	password := d.Get("password").(string)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, jaId.ResourceGroup, jaId.ServerName, jaId.Name, name)
```
Comment (jackofallops) [inline azurerm/internal/services/mssql/mssql_job_credential_resource.go:80]:
```suggestion
		existing, err := client.Get(ctx, jobCredentialId.ResourceGroup, jobCredentialId.ServerName, jobCredentialId.JobAgentName, jobCredentialId.CredentialName)
```

File: azurerm/internal/services/mssql/mssql_job_credential_resource.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Failed to check for presence of existing MsSql Job Credential %q (Job Agent %q / MsSql Server %q / Resource Group %q): %s", name, jaId.Name, jaId.ServerName, jaId.ResourceGroup, err)
```
Comment (jackofallops) [inline azurerm/internal/services/mssql/mssql_job_credential_resource.go:83]:
```suggestion
				return fmt.Errorf("checking for presence of existing MsSql %s: %+v", jobCredentialId, err)
```

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @r0bnet - Many thanks for this PR, looking mostly good, but I've left a few suggestions for making use of the generated ID code.  We're aiming to use these throughout the provider, so newly introduced resources can make use of them on Create (we're working through existing resources slowly as they will need state migrations, hence largely looking like only in Read).
Thanks again!

File: azurerm/internal/services/mssql/mssql_job_credential_resource.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+			Password: utils.String(password),
+		},
+	}
+
+	cred, err := client.CreateOrUpdate(ctx, jaId.ResourceGroup, jaId.ServerName, jaId.Name, name, jobCredential)
+	if err != nil {
+		return fmt.Errorf("creating MsSql Job Credential %q (Job Agent %q / Sql Server %q / Resource Group %q): %+v", name, jaId.Name, jaId.ServerName, jaId.ResourceGroup, err)
+	}
+
+	d.SetId(*cred.ID)
```
[the remark this answers is not in the document]
Comment (r0bnet) [reply azurerm/internal/services/mssql/mssql_job_credential_resource.go:105]: Yeah, sure! :)

**Comment by r0bnet** [comment]
Thanks @jackofallops. I committed your suggestions. Hope it's good to go now 👍

**Review by tombuildsstuff — APPROVED** [review]
LGTM - thanks for pushing those changes @r0bnet

**Pull request merged**